### ISSUE TITLE
Show users available checkpoints

### ISSUE THREAD

**Issue by jdeschamps** [CAREamics/careamics#305]
## Problem

Once a model is trained, users have to inspect the checkpoints folder to know what has been saved. There is no indication of the loss or metrics corresponding to the validation set and the particular checkpoint.

The `CAREamist` should be able to print this information to users, it has access to the loss (and metrics in the future) for each epoch. Then it could also allow users to choose a checkpoint and load it, via maybe an `epoch_number`, rather than have them load manually via its path.
